packages/cli-repl/test/e2e-auth.spec.ts

```diff
@@ -930,7 +936,10 @@ describe('Auth e2e', function () {
     });
     context('with specific auth mechanisms', function () {
       it('can auth with SCRAM-SHA-1', async function () {
-        if (process.env.MONGOSH_TEST_E2E_FORCE_FIPS) {
+        if (
+          process.env.MONGOSH_TEST_E2E_FORCE_FIPS ||
+          process.env.DISTRO_ID === 'rhel92-fips'
```

Comment:
Why do we need to check the DISTRO_ID? What would be the impact of, at the CI level, specifying MONGOSH_TEST_E2E_FORCE_FIPS=true on the rhel92-fips variant?

Reply:
`MONGOSH_TEST_E2E_FORCE_FIPS` makes our test runner for the e2e tests always append the `--tlsFIPSMode` flag to the mongosh command line, so mongosh always runs in FIPS mode.

The `rhel92-fips` machines seem to be configured in a similar way; their system OpenSSL configuration always runs everything in FIPS mode (this seems to be different from `rhel83-fips`, by the way; it might be a question of OpenSSL 1.1 vs OpenSSL 3).

If we set `MONGOSH_TEST_E2E_FORCE_FIPS=true` here, we wouldn’t be testing the behavior of mongosh on a FIPS-enforcing machine without the `--tlsFIPSMode` flag anymore. That might be okay, but I think I’d mildly prefer to have the extra test coverage here, even if that means running two e2e test tasks with the same effective behavior.

Reply:
OK that is a huge detail I wasn't aware of. Thanks a lot for the explanation!